FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -206,47 +226,149 @@ await foreach (FhirToDataLakeProcessingJobInputData input in inputs.WithCancella
+                        _logger.LogInformation($"One small job is generated and pushed into merge list for {resourceType} with {subJob.ResourceCount} count.");
+
+                        var jobParameters = PushToJobPool(resourceType, subJob.TimeRange, subJob.ResourceCount);
+                        if (jobParameters != null)
+                        {
+                            _logger.LogInformation($"Generated one merged job with {jobParameters.Keys} types.");
+
+                            yield return new FhirToDataLakeProcessingJobInputData
+                            {
+                                JobType = JobType.Processing,
+                                ProcessingJobSequenceId = _result.CreatedJobCount,
+                                TriggerSequenceId = _inputData.TriggerSequenceId,
+                                Since = _inputData.Since,
+                                SplitParameters = jobParameters,
+                            };
+                        }
+                    }
+                    else
+                    {
+                        _logger.LogInformation($"Generated one job for {resourceType} with {subJob.ResourceCount} count.");
+                        yield return new FhirToDataLakeProcessingJobInputData
+                        {
+                            JobType = JobType.Processing,
+                            ProcessingJobSequenceId = _result.CreatedJobCount,
+                            TriggerSequenceId = _inputData.TriggerSequenceId,
+                            Since = _inputData.Since,
+                            SplitParameters = new Dictionary<string, TimeRange>() { { subJob.ResourceType, subJob.TimeRange } },
+                        };
+                    }
+                }
+            }
+
+            if (_mergeList.Count != 0)
+            {
+                var jobParameters = _mergeList.ToDictionary(x => x.Item1, x => x.Item2);
+
+                _logger.LogInformation($"Generated one merged job with {jobParameters.Keys} types.");
+                yield return new FhirToDataLakeProcessingJobInputData
                 {
                     JobType = JobType.Processing,
                     JobVersion = _inputData.JobVersion,
                     ProcessingJobSequenceId = _result.CreatedJobCount,
                     TriggerSequenceId = _inputData.TriggerSequenceId,
                     Since = _inputData.Since,
-                    DataStartTime = lastEndTime,
-                    DataEndTime = nextJobEnd,
+                    SplitParameters = jobParameters,
                 };
-                _result.NextJobTimestamp = nextJobEnd;
-                yield return input;
             }
         }
 
+        private Dictionary<string, TimeRange> PushToJobPool(string resourceType, TimeRange range, int count)
+        {
+            // Push small job into pool list.
+            var currentCount = _mergeList.Sum(x => x.Item3);
```

Comment:
Looks like this is more of a candidate job instead of a job pool? Shall we declare some `JobCandidate { int count; dictionary<string, range> }` so that we can process the small jobs more directly.